gluten-core/src/main/scala/org/apache/gluten/config/GlutenConfig.scala

```diff
@@ -117,6 +117,9 @@ class GlutenConfig(conf: SQLConf) extends Logging {
   def enableExtendedColumnPruning: Boolean =
     getConf(ENABLE_EXTENDED_COLUMN_PRUNING)
 
+  def enableEnhancedFeature(): Boolean =
+    System.getenv().getOrDefault("GLUTEN_ENABLE_ENHANCED_FEATURE", "false").toBoolean
+
```

Comment:
Should this be defined in `GlutenConfig.scala` or in `VeloxConfig.scala`?

And it feels worth it to add an individual JNI method to avoid relying on the env?

E.g.,

```java
native boolean isEnhancedFeaturesEnabled();
```

More generally in the future, we may have

```java
native String[] getEnabledEnhancedFeatures();
```

Reply:
@jinchengchenghh Got it. I think we need a backend API for the feature then. E.g., `BackendSettingsApi#isEnhancedFeatureEnabled()`

For CH backend, we can make it false by default.

Then we can move `ConfigJniWrapper.java` which I see is newly added to `gluten-core` to `backends-velox`. This is to follow the rule that we don't put any JNI code to `gluten-core` so it's more flexible for backends.